File: tests/machine/x/clj/user_type_literal.clj
(ns main)

(declare book)

(defn Person [name age]
  {:__name "Person" :name name :age age}
)


(defn Book [title author]
  {:__name "Book" :title title :author author}
)


(defn -main []
  (def book {:__name "Book" :title "Go" :author {:__name "Person" :name "Bob" :age 42}}) ;; Book
  (println (:name (:author book)))
)

(-main)

File: tests/machine/x/fs/user_type_literal.fs
open System

type Person = {
    mutable name: string
    mutable age: int
}
type Book = {
    mutable title: string
    mutable author: Person
}
let book: obj = { title = "Go"; author = { name = "Bob"; age = 42 } }
printfn "%A" (book.author.name)
